src/Uno.UI/UI/Xaml/Controls/ComboBox/ComboBox.cs

```diff
@@ -250,36 +243,59 @@ private void UpdateContentPresenter()
 		{
 			if (_contentPresenter != null)
 			{
-				var item = GetSelectionContent();
-
-				var itemView = item as _View;
-
-				if (itemView != null)
+				if (SelectedItem != null)
 				{
+					var item = GetSelectionContent();
+					if (item is _View itemView)
+					{
 #if __ANDROID__
-					var comboBoxItem = itemView.FindFirstParentOfView<ComboBoxItem>();
+						var comboBoxItem = itemView.FindFirstParentOfView<ComboBoxItem>();
```

Comment:
`SelectedItem` may *not* be a `ComboBoxItem` though, in fact it usually won't be. Also it's not an ironclad guarantee that `ContentControl.Content` will be a child of `ContentControl`.

Reply:
I tried to avoid walking up the tree when the item is ComboBoxItem:
```cs
        private ComboBoxItem GetSelectionContainer()
        {
            if (SelectedItem is ComboBoxItem item)
            {
                return item;
            }
            else if (SelectedItem is _View view)
            {
#if __ANDROID__
                return view.FindFirstParentOfView<ComboBoxItem>();
#else
                return view.FindFirstParent<ComboBoxItem>();
#endif
            }
            else
            {
                return null;
            }
        }
        private object GetSelectionContent()
        {
            return SelectedItem is ComboBoxItem cbi ? cbi.Content : SelectedItem;
        }
//...
var item = GetSelectionContent();
if (item is _View itemView)
{
   var comboBoxItem = itemView.FindFirstParentOfView<ComboBoxItem>(); // old impl
   var comboBoxItem2 = GetSelectionContainer(); // new impl attempt
```

However that lead to some issue when the item got moved under ContentPresenter in the visual tree. 
Particularly, in the "ComboBox_ComboBoxItem_Sel..." sample, when "item 1" is selected and you open the flyout again, the item 1 is not restored. (Item 2-4 didn't have this problem.) What I noticed is that `UpdateContentPresenter` was called twice, and on the second call comboBoxItem is null, while comboBoxItem2 has value. Maybe this corrupts the `_selectionParentInDropdown`, thus prevent the item in the dropdown being restored.

I am leaving the code as it is now.